Angular/src/app/dashboard/bibliography/bibliography.component.ts

```diff
@@ -1,209 +1,100 @@
-import { Component, Input, OnInit } from '@angular/core';
-import { FormArray, FormControl, FormGroup } from '@angular/forms';
-import { MatTabChangeEvent } from '@angular/material/tabs';
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { ApiService } from '@oscc/api.service';
 import { AuthService } from '@oscc/auth/auth.service';
 import { DialogService } from '@oscc/services/dialog.service';
-import { Observable, map, startWith } from 'rxjs';
+
+import { ViewChild } from '@angular/core';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+
+import { Bib } from '@oscc/models/Bib';
 
 @Component({
   selector: 'app-bibliography',
   templateUrl: './bibliography.component.html',
   styleUrls: ['./bibliography.component.scss'],
 })
-export class BibliographyComponent implements OnInit {
-  retrieved_bibliography_authors: object;
-  retrieved_author_bibliography: object;
-
-  bibliography_form = new FormGroup({
-    _id: new FormControl(''),
-    bib_entry_type: new FormControl('book'), // Book is default on page load
-    author: new FormControl(''),
-    title: new FormControl(''),
-    year: new FormControl(''),
-    series: new FormControl(''),
-    number: new FormControl(''),
-    location: new FormControl(''),
-    edition: new FormControl(''),
-    journal: new FormControl(''),
-    volume: new FormControl(''),
-    pages: new FormControl(''),
-  });
-
-  // Bibliography author selection
-  bibliography_author_selection_form = new FormControl();
-  bibliography_author_selection_form_options: string[] = [];
-  bibliography_author_selection_form_filtered_options: Observable<string[]>;
-  bibliography_form_selected_type = new FormControl(0);
-  bib_entry_selected = false;
-
-  constructor(protected auth_service: AuthService, protected api: ApiService, protected dialog: DialogService) {}
+export class BibliographyComponent implements OnInit, OnDestroy {
+  @ViewChild('paginator') paginator: MatPaginator;
+  @ViewChild('sort') sort: MatSort;
+  table_columns: string[] = ['name', 'title', 'date'];
+  table_data: MatTableDataSource<any>;
+  table_source_data: any[];
+
+  private bib_subscription: any;
+  protected pages: string;
+
+  constructor(protected auth_service: AuthService, protected api: ApiService, protected dialog: DialogService) {
+    this.table_data = new MatTableDataSource(this.table_source_data);
+  }
 
   /**
    * On Init, we just load the list of authors. From here, selection is started
```

Comment:
Dit is vgm niet langer t geval

Reply:
_this.table_data is not defined_ als je dit niet doet :D